src/rosdistro/manifest_provider/git.py

```diff
@@ -47,102 +49,76 @@
 def git_manifest_provider(_dist_name, repo, pkg_name):
     assert repo.version
     try:
-        release_tag = get_release_tag(repo, pkg_name)
-        package_xml = _get_package_xml(repo.url, release_tag)
-        return package_xml
+        release_tag = repo.get_release_tag(pkg_name)
+        with _temp_git_clone(repo.url, release_tag) as git_repo_path:
+            filename = os.path.join(git_repo_path, 'package.xml')
+            if not os.path.exists(filename):
+                raise RuntimeError('Could not find package.xml in repository "%s"' % repo.url)
+            with open(filename, 'r') as f:
+                return f.read()
     except Exception as e:
         raise RuntimeError('Unable to fetch package.xml: %s' % e)
 
 
-def _git_version_gte(version):
-    global _git_client_version
-    if not _git_client_version:
-        cmd = [_git_client_executable, '--version']
-        result = _run_command(cmd)
-        _git_client_version = result['output'].split()[-1]
-    return LooseVersion(_git_client_version) >= LooseVersion(version)
+def git_source_manifest_provider(repo):
+    try:
+        with _temp_git_clone(repo.url, repo.version) as git_repo_path:
+            # Include the git hash in our cache dictionary.
+            result = Git(git_repo_path).command('rev-parse', 'HEAD')
+            cache = { '_ref': result['output'] }
+
+            for path, dirnames, filenames in os.walk(git_repo_path):
```

Comment:
This loop is reimplementing `catkin_pkg` functionality and e.g. doesn't honor `CATKIN_IGNORE` files, doesn't check for duplicate package names.

Can this reuse [catkin_pkg.packages.find_package_paths](https://github.com/ros-infrastructure/catkin_pkg/blob/master/src/catkin_pkg/packages.py#L41) or [catkin_pkg.packages.find_packages](https://github.com/ros-infrastructure/catkin_pkg/blob/master/src/catkin_pkg/packages.py#L71)?